docs/source-2.0/aws/rules-engine/library-functions.rst

```diff
@@ -0,0 +1,237 @@
+functions:
+
+The following example adds ``smithy-aws-rules-engine`` as a Gradle dependency
+to a ``build.gradle.kts`` file:
+
+.. code-block:: kotlin
+
+    dependencies {
+        ...
+        implementation("software.amazon.smithy:smithy-aws-rules-engine:__smithy_version__")
+        ...
+    }
+
+.. _rules-engine-aws-library-awsPartition:
+
+``aws.partition`` function
+==========================
+
+Summary
+    Provides a `Partition structure`_ of content for the
+Argument type
+    * region: ``string``
+Return type
+    ``option<Partition>``
+
+    *Contains the region's partition, or an empty optional if the region is
+    unknown.*
+
+The following example uses ``aws.partition`` to partition information from the
+region in the value of the ``foo`` parameter:
+
+.. code-block:: json
+
+    {
+        "fn": "aws.partition",
+        "argv": [
+            {"ref": "foo"}
+        ]
+    }
+
+
+.. _rules-engine-aws-library-awsPartition-Partition:
+
+-----------------------
+``Partition`` structure
+-----------------------
+
+The ``Partition`` structure is returned from the `aws.partition function`_ when
+its input is a known region. The ``Partition`` object contains the following
+properties:
+
+    .. list-table::
+        :header-rows: 1
+        :widths: 10 20 70
+
+        * - Property
+          - Type
+          - Description
+        * - name
+          - ``string``
+          - The partition's name.
+        * - dnsSuffix
+          - ``string``
+          - The partition's default DNS suffix.
+        * - dualStackDnsSuffix
+          - ``string``
+          - The partition's dual-stack specific DNS suffix.
+        * - supportsFIPS
+          - ``bool``
+          - Indicates whether the partition supports a FIPS compliance mode.
+        * - supportsDualStack
+          - ``bool``
+          - Indicates whether the partition supports dual-stack endpoints.
+
+
+.. _rules-engine-aws-library-awsParseArn:
+
+``aws.parseArn`` function
+=========================
+
+Summary
+    Computes an `ARN structure`_ given an input ``string``.
+Argument type
+    * value: ``string``
+Return type
+    ``option<ARN>``
+
+    *Contains the parsed ARN, or an empty optional if the ARN could not be
+    parsed*
+
+The following example uses ``aws.parseArn`` to parse the value of the ``foo``
+parameter into its component parts:
+
+.. code-block:: json
+
+    {
+        "fn": "aws.parseArn",
+        "argv": [
+            {"ref": "foo"}
+        ]
+    }
+
+
+.. _rules-engine-aws-library-parseArn-Arn:
+
+-----------------
+``ARN`` structure
+-----------------
+
+The ``ARN`` structure is returned from the `aws.parseArn function`_ when its
+input is a valid ARN. The ``ARN`` object contains the following properties:
+
+    .. list-table::
+        :header-rows: 1
+        :widths: 10 20 70
+
+        * - Property
+          - Type
+          - Description
+        * - partition
+          - ``string``
+          - The partition where the resource is located.
```

Comment:
Should we link to https://docs.aws.amazon.com/whitepapers/latest/aws-fault-isolation-boundaries/partitions.html?